engine-tests/build.gradle

```diff
@@ -58,15 +58,16 @@ dependencies {
     implementation group: 'org.reflections', name: 'reflections', version: '0.9.10'
 
     // Test lib dependencies
-    implementation("org.junit.jupiter:junit-jupiter-api:5.5.2")
-    implementation("org.junit.jupiter:junit-jupiter-params:5.5.2")
-    implementation group: 'org.mockito', name: 'mockito-junit-jupiter', version: '3.2.0'
-
-    implementation("org.terasology.joml-ext:joml-test:1.0.0-SNAPSHOT")
+    implementation("org.junit:junit-bom:5.7.1") {
+        // junit-bom will set version numbers for the other org.junit dependencies.
+    }
+    implementation("org.junit.jupiter:junit-jupiter-api")
+    implementation("org.junit.jupiter:junit-jupiter-params")
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine")
 
-    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.5.2")
+    implementation("org.mockito:mockito-junit-jupiter:3.7.7")
 
-    implementation group: 'junit', name: 'junit', version: '4.12'
```

Comment:
I did remove this dependency on old JUnit 4 from engine-test. I expect it to fail loudly if that was a bad idea.

(JUnit 4 _is_ still in the terasology-modules deps in case any of the modules haven't converted yet.)